### ISSUE TITLE
Error checking for linux updates: Invalid latest version format:

### ISSUE THREAD

**Issue by philrz** [brimsec/brim#1219]
On an Ubuntu Linux VM, I happened to start my Brim GA `v0.19.0` app by executing `brim` in a shell. It started fine, but it showed the error message:

```
12:23:18.726 > Error checking for linux updates: Invalid latest version format:
```

I'm suspecting this might actually be a benign side effect of there being no newer version available. However, it'd be good to know for sure.

I've marked this as a `v0.20.0` target just to make sure we give it a look. If someone looks at the code and decides it's benign, we can surely put it off addressing it until later.

**Comment by philrz** [comment]
I've verified the original error message is no longer shown on Linux, using a scratch Brim release candidate based on commit `e0c7d24` that includes the change in #1244. Now when I startup Brim via the shell in Ubuntu 18.04, the error message is no longer present:

```
$ brim
08:56:56.505 › app paths: getAppPath=/usr/lib/brim/resources/app userData=/home/phil/.config/Brim logs=/home/phil/.config/Brim/logs
08:56:56.636 › migrations: currentVersion=202012011232 pending=0
08:56:56.640 › zqd core log /home/phil/.config/Brim/logs/zqd-core.log
08:56:56.641 › zqd access log /home/phil/.config/Brim/logs/zqd-access.log
08:56:56.641 › zqd config /home/phil/.config/Brim/zqd-config.yaml
08:56:56.646 › spawning zqd: /usr/lib/brim/resources/app/zdeps/zqd listen -l localhost:9867 -data /home/phil/.config/Brim/data/spaces -config /home/phil/.config/Brim/zqd-config.yaml -suricatarunner /usr/lib/brim/resources/app/zdeps/suricata/suricatarunner -suricataupdater /usr/lib/brim/resources/app/zdeps/suricata/suricataupdater -zeekrunner /usr/lib/brim/resources/app/zdeps/zeek/zeekrunner -brimfd=3
zqd started on:  localhost:9867
init: 1161.890ms
```

There's more significant positive user impact from the additional changes that came with #1244, per this quote:

> if the latest version is ahead of the current version then don't update. 

I've verified part of this by comparing the behavior of a scratch Brim release candidate based on commit `e0c7d24` that includes this change vs. another one based on older commit `6bb51cb` that does not. Both have version string `v0.21.0`. Running the `6bb51cb` one on macOS, after waiting just over a minute post-launch, I was presented with the pop-up informing me that the app had auto-updated itself to `v0.20.0` (which is the latest current GA release) and prompting me to update now or later. By comparison, with the `e0c7d24` one that includes the changes in #1244, I launched the app and let it hang around on my desktop for 40+ minutes and never saw any signs of auto-update having fired, which is what we'd now expect since the latest version is ahead of the current version.

Thanks @mason-fish!

**Comment by philrz** [comment]
I've done a round of more formal verification now that Brim GA releases tagged `v0.21.0` and `v0.21.1` are both available and `v0.21.1` is the newest available that autoupdate would check against.

On each platform, I've started from a fresh install of Brim GA tagged `v0.21.0` and launched the app post-install, with the "About" window shown in the screenshot as confirmation. In each case, within a couple minutes, a pop-up shows that the autoupdate confirmed the newer version is available. In the macOS and Windows cases, the update is automatic and the prompt is to restart into the updated version, and on Linux it's just a pointer to download the new version (see https://github.com/brimsec/brim/issues/871).

### macOS

![image](https://user-images.githubusercontent.com/5934157/103553735-b7106a80-4e62-11eb-92a8-6d08f4d68f68.png)

### Windows

![image](https://user-images.githubusercontent.com/5934157/103554333-998fd080-4e63-11eb-9e52-176c2b177e53.png)

### Linux

![image](https://user-images.githubusercontent.com/5934157/103554675-25096180-4e64-11eb-87b2-c8f9b6a9f7de.png)

Perhaps of more significance is how this logic improves our ability to offer "pre-release" builds to users who volunteer to be early testers on new fixes, or "hotfix" releases that include bug fixes targeted at specific users. Two key considerations regarding this:

1. We want to be able to set specific version strings in these releases so users will see them in the "About" menu when they report issues to us

2. We'd typically want these releases to still autoupdate to the next tagged GA release when it comes out, since any changes we'd made available via pre-release/hotfix builds would hopefully be included in the next GA release

To test this, I created a pair of scratch builds based off a recent Brim `master` commit (38a5e69) each with the only change being to the `version` string in `package.json` and `package-lock.json`. These modified version strings adhered to section 9 of the [Semantic Versioning spec](https://semver.org/) in that were extended with only a hyphen and the word `prerelease`, one build using the string `0.21.1-prerelease` (so, the base version number matches the latest currently available) and the other using `0.22.0-prerelease` (so, the base version number in that string is "newer" than the latest currently available).

As would be expected, if starting from an install with the build tagged `0.21.1-prerelease`, we once again see the autoupdate is triggered. This verifies the expected functionality of a user that had been running a prerelease build we'd given them and the GA release that delivers the equivalent change/fix has now come out.

### macOS

![image](https://user-images.githubusercontent.com/5934157/103556162-37849a80-4e66-11eb-9a15-a4c6906189b3.png)

### Windows

![image](https://user-images.githubusercontent.com/5934157/103556707-05c00380-4e67-11eb-8c0b-042df2a7ec86.png)

### Linux

![image](https://user-images.githubusercontent.com/5934157/103557444-115ffa00-4e68-11eb-9010-1a5e4c228500.png)

Then for the build tagged `0.22.0-prerelease`, we can see that after having been launched for several minutes, no autoupdate has been triggered. This verifies the expected functionality of a user that needs to keep running a "current" prerelease/hotfix build until the GA build with tag `0.22.0` is released.

### macOS

![image](https://user-images.githubusercontent.com/5934157/103559408-edea7e80-4e6a-11eb-88c7-c917a74a2597.png)

### Windows

![image](https://user-images.githubusercontent.com/5934157/103574529-f94aa380-4e84-11eb-8d5d-74ee743cb979.png)

### Linux

![image](https://user-images.githubusercontent.com/5934157/103558418-7cf69700-4e69-11eb-8c25-861991a31f49.png)

The one supportability use case I can see that we're not yet able to cover is that of a user that has a prerelease/hotfix build that _doesn't_ want to be autoupdated to the next GA release when it comes out (i.e. a user that's dependent on functionality that's going to take a couple release cycles to go GA). For this I've expected we'll eventually want to offer an option in the Preferences menu so the user can opt out of autoupdate checks, and that's tracked in #1211.
